features/prompt-registry.mdx

```diff
@@ -258,13 +368,67 @@ response = openai.ChatCompletion.create(
 )
 ```
 
+```js JavaScript
+const messages = [
+  { role: "user", content: "What's the weather like in Boston?" },
+];
+const functions = [
+  {
+    name: "get_current_weather",
+    description: "Get the current weather in a given location",
+    parameters: {
+      type: "object",
+      properties: {
+        location: {
+          type: "string",
+          description: "The city and state, e.g. San Francisco, CA",
+        },
+        unit: { type: "string", enum: ["celsius", "fahrenheit"] },
+      },
+      required: ["location"],
+    },
+  },
+];
+
+promptlayer.prompts.publish({
+  prompt_name: "test_chat_functions",
+  prompt_template: {
+    function_call: "auto", // or {"name": function_name} if you want to make sure the function gets called.
+    input_variables: [], // list them here if it has any
+    messages: messages.map(({ role, content }) => ({
+      role,
+      prompt: {
+        input_variables: [], // list them here if it has any
+        template: content,
+        template_format: "f-string", // or "jinja"
+        validate_template: true, // optional
```

Comment:
What is validate_template??

Reply:
@jzone3 this is coming from langchain. Should we remove this? We are eventually going to remove our dependency on langchain.

Reply:
I hate it but good point lol